corehq/apps/hqwebapp/templatetags/tests/test_tag.py

```diff
@@ -71,6 +72,66 @@ def test_initial_page_data(self):
             ],
         })
 
+    def test_html_attr_string(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr "circus" %}"""),
+            "circus"
+        )
+
+    def test_html_attr_string_with_quotes(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr what_on_earth %}""", {
+                "what_on_earth": """cir'cus ci"rcus""",
+            }),
+            "cir&#x27;cus ci&quot;rcus",
+        )
+
+    def test_html_attr_translated_string(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr _("toxic") %}"""),
+            "toxic"
+        )
+
+    def test_html_attr_number(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr "23" %}"""),
+            "23"
+        )
+
+    def test_html_attr_bool(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr True %}"""),
+            "true"
+        )
+
+    def test_html_attr_None(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr None %}"""),
+            "null"
+        )
+
+    def test_html_attr_list(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr my_list %}""", {
+                "my_list": ["circus", "toxic", "stronger"],
+            }),
+            "[&quot;circus&quot;, &quot;toxic&quot;, &quot;stronger&quot;]"
+        )
+
+    def test_html_attr_dict(self):
+        self.assertEqual(
+            self.render("""{% load hq_shared_tags %}{% html_attr my_dict %}""", {
+                "my_dict": {"toxic": 2003, "circus": 2008}
+            }),
+            "{&quot;toxic&quot;: 2003, &quot;circus&quot;: 2008}"
+        )
+
+    def test_html_attr_obj(self):
+        with self.assertRaises(TemplateTagJSONException):
+            self.render("""{% load hq_shared_tags %}{% html_attr my_object %}""", {
+                "my_object": object(),
+            })
+
```

Comment:
I'm pretty sure this will fail:
```suggestion

    def test_html_attr_escape_translated_string(self):
        self.assertEqual(
            self.render("""{% load hq_shared_tags %}{% html_attr _('this is ">dangerous') %}"""),
            "this is &quot;&gt;dangerous"
        )
```